### PR TITLE
Fix blank page for buy orders wrapEth

### LINKED ISSUE #557: CowSwap shows a blank page
**Describe the bug**
Reported by a user on Discord:

```
cowswap shows a blank page when I did the following:
-Selected a token BAL
-Put 1 BAL
-Change wETH to ETH.

The javascript console shows an error: Invariant Failed.
```

**How to Reproduce**

See above

**Expected behavior**
The app should not crash

**Screenshots**

**Additional context**
Also, if it's relevant for the described bg, add information like the browser or OS.

### PR DESCRIPTION
**Pull request by anxolin**
Closes #557 

Makes the app not to break for buy orders where ETH is the sell token.

It doesn't include the ETH wrapper tool, for that #560 

The issue was basically, it was using the typed amount instead of always the input for doing the check of low balance. Then it was operating using amounts of two tokens that were not related (ETH and DAI), so it was breaking the invariant.

## Test
Try to type a buy order with ETH as the sell token. You won't seel the ETH wrapper message/tool, but the app won't break either. 

@W3stside this needs careful testing. I need to jump into a call now, but I appreciate if you can bullet proof test these two prs.

### CODE REVIEW

File: src/custom/pages/Swap/SwapMod.tsx [hunk 1]
```diff
@@ -471,11 +471,11 @@ export default function Swap({
                   )}
                   {isFeeGreater && fee && <FeeGreaterMessage fee={fee} />}
                 </AutoColumn>
-                {isNativeIn && onWrap && (
+                {isNativeIn && trade && onWrap && (
```
Comment (W3stside) [inline src/custom/pages/Swap/SwapMod.tsx:474]: not technically necessary but good to check trade here

**Comment by W3stside** [comment]
@anxolin do we really want to use the `inputAmountWithFee`? it's not what the exactIn input uses (inputAmount), see below:
Screenshot from 2021-04-29 11-31-49

**Comment by anxolin** [comment]
Nup, you are right. We should use what he inputs (for sell orders) and the amount after fees for buy ones. Right?

do you mind making the amendment PR on top of this?

**Comment by W3stside** [comment]
concern: https://github.com/gnosis/gp-swap-ui/pull/568#issuecomment-829946824

**Comment by W3stside** [comment]
Can confirm that MM shows truncated amounts but in etherscan it does indeed use the full 18 precision, can be tested by selling `0.19004712312312312` ETH for some token, when wrapping check MM, then check the etherscan resolved tx

## Setup, MM showing truncated, `From` field showing full
Screenshot from 2021-04-30 10-15-20

## Etherscan confirmed tx:
Screenshot from 2021-04-30 10-15-44

File: src/custom/pages/Swap/SwapMod.tsx (hunk 1 above)
Comment (W3stside) [reply src/custom/pages/Swap/SwapMod.tsx]: this has been changed as it doesn't add anything.

**Comment by anxolin** [comment]
Awesome!

**Comment by W3stside** [comment]
Approving verbally as apparently i can't approve this PR
Screenshot from 2021-04-30 11-30-32

**Comment by anxolin** [comment]
you have broken Github @W3stside

**Pull request merged**